rust/operator-binary/src/crd/mod.rs

```diff
@@ -542,6 +546,24 @@ impl v1alpha1::HbaseCluster {
         }
     }
 
+    /// Returns required metrics port name and metrics port number tuples depending on the role.
+    /// The metrics are available over the UI port.
+    pub fn metrics_ports(&self, role: &HbaseRole) -> Vec<(String, u16)> {
+        match role {
+            HbaseRole::Master => vec![(
+                HBASE_METRICS_PORT_NAME.to_string(),
+                HBASE_MASTER_METRICS_PORT,
+            )],
+            HbaseRole::RegionServer => vec![(
+                HBASE_METRICS_PORT_NAME.to_string(),
+                HBASE_REGIONSERVER_METRICS_PORT,
+            )],
+            HbaseRole::RestServer => {
+                vec![(HBASE_METRICS_PORT_NAME.to_string(), HBASE_REST_METRICS_PORT)]
```

Comment:
suggestion: Use `to_owned` instead.

```suggestion
                HBASE_METRICS_PORT_NAME.to_owned(),
                HBASE_MASTER_METRICS_PORT,
            )],
            HbaseRole::RegionServer => vec![(
                HBASE_METRICS_PORT_NAME.to_owned(),
                HBASE_REGIONSERVER_METRICS_PORT,
            )],
            HbaseRole::RestServer => {
                vec![(HBASE_METRICS_PORT_NAME.to_owned(), HBASE_REST_METRICS_PORT)]
```